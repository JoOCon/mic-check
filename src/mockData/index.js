export const mockUser = {
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
};

export const mockUsers = [{
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
},
{
  userName: 'Jim Hanks',
  password: 'password',
  email: 'JHanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
}];

export const mockItem = {
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
};

export const mockItem2 = {
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
};

export const mockItems = [{
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
}, 
{
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
}];

export const mockSenderLocation = '101 Market St, San Francisco, CA';
export const mockDeliveryLocation = '20 McAllister St, San Francisco, CA';

export const mockQuoteData = {
  kind: "delivery_quote",
  fee: 1140,
  created: "2018-09-13T00:40:02Z",
  pickup_duration: 5,
  expires: "2018-09-13T00:45:01Z",
  currency: "usd",
  currency_type: "USD",
  duration: 45,
  dropoff_eta: "2018-09-13T01:25:02Z",
  id: "dqt_Ltalsge5ktl1YV"
};

export const mockOrderInfo = {
  manifest: 'a box of shoes',
  'manifest_items[].name': 'Nikes',
  'manifest_items[].size': '2',
  pickup_address: '101 Market St, San Francisco, CA',
  pickup_name: 'Allen',
  pickup_phone_number: 3033334444,
  dropoff_address: '20 McAllister St, San Francisco, CA',
  dropoff_name: 'Tim',
  dropoff_phone_number: 2223334444,
  dropoff_notes: 'Tim@timmmy.com'
};