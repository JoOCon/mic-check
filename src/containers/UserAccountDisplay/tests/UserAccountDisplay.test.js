import React from 'react';
import { shallow } from 'enzyme';
import { UserAccountDisplay, mapStateToProps } from '../';
import * as mockData from '../../../mockData';

describe('UserAccountDisplay tests', () => {
  let wrapper;
  const { mockUser, mockOrderConfirmation } = mockData;
  beforeEach(() => {
    wrapper = shallow( 
      <UserAccountDisplay 
        completedOrders={[mockOrderConfirmation]}
        activeUser={mockUser}
      /> 
    );
  });

  it('renders the UserAccountDisplay with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should have a completedOrders data object in props', () => {
      const mockState = {completedOrders: [mockOrderConfirmation]};
      const expected = {completedOrders: [mockOrderConfirmation]};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });

    it('should have a activeUser data object in props', () => {
      const mockState = {activeUser: mockUser};
      const expected = {activeUser: mockUser};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });
});
