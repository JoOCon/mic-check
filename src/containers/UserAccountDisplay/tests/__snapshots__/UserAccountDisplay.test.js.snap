// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccountDisplay tests renders the UserAccountDisplay with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserAccountDisplay
    activeUser={
        Object {
            "email": "THanks@hanks.com",
            "location": "20 McAllister St, San Francisco, CA",
            "password": "password",
            "userName": "Tom Hanks",
          }
    }
    completedOrders={
        Array [
            Object {
              "complete": false,
              "courier": null,
              "courier_imminent": false,
              "created": "2018-09-17T22:20:07Z",
              "currency": "usd",
              "dropoff": Object {
                "name": "Allen",
                "notes": "Tim@timmmy.com",
                "phone_number": 3033334444,
              },
              "dropoff_deadline": "2018-09-17T22:50:07Z",
              "dropoff_eta": null,
              "dropoff_identifier": "",
              "dropoff_ready": "2018-09-17T22:30:07Z",
              "fee": 571,
              "id": "del_Lu-c_oyiH69mDF",
              "kind": "delivery",
              "live_mode": false,
              "manifest": Object {
                "description": "a box of shoes",
              },
              "pickup": Object {
                "name": "Tim",
                "notes": "",
                "phone_number": 2223334444,
              },
              "pickup_deadline": "2018-09-17T22:40:07Z",
              "pickup_eta": null,
              "pickup_ready": "2018-09-17T22:30:07Z",
              "quote_id": "dqt_Lu-c_vlL5PdvEF",
              "related_deliveries": Array [],
              "status": "pending",
              "tip": null,
              "tracking_url": "https://postmates.com/track/del_Lu-c_oyiH69mDF",
              "undeliverable_action": null,
              "undeliverable_reason": null,
              "updated": "2018-09-17T22:20:07Z",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "User Orders",
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      "User Orders",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "User Orders",
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        "User Orders",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
